Transient Test Error - test_delete_item_not_in_cache

FAILED test/integration/objectstore/test_remote_objectstore_cache_operations.py::TestCacheOperation::test_delete_item_not_in_cache - requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://127.0.0.1:9664/api/histories/529fd61ab1c6cc36/contents/529fd61ab1c6cc36


```
    def test_delete_item_not_in_cache(self):
        hda = self.upload_dataset()
        assert files_count(self.object_store_cache_path) == 1
        shutil.rmtree(self.object_store_cache_path)
        os.mkdir(self.object_store_cache_path)
        assert files_count(self.object_store_cache_path) == 0
>       self.dataset_populator.delete_dataset(hda["history_id"], hda["id"], purge=True, wait_for_purge=True)

```

Server Side Logging:

```
INFO:     127.0.0.1:34178 - "GET /api/histories/529fd61ab1c6cc36/contents/529fd61ab1c6cc36 HTTP/1.1" 200 OK
INFO:     127.0.0.1:34184 - "GET /api/histories/529fd61ab1c6cc36/contents/529fd61ab1c6cc36 HTTP/1.1" 200 OK
INFO:     127.0.0.1:34198 - "GET /api/histories/529fd61ab1c6cc36/contents/529fd61ab1c6cc36 HTTP/1.1" 200 OK
celery.worker.request DEBUG 2025-12-24 23:59:47,041 [pN:main,p:8250,tN:Thread-31] Task accepted: galaxy.celery.tasks.purge_hda[c2a8c5a7-98ff-46cb-9a86-c6516ee1e5fe] pid:8250
galaxy.objectstore.s3 DEBUG 2025-12-24 23:59:47,078 [pN:main,p:8250,tN:Thread-31] Deleting key a/a/0/dataset_aa0b1f6e-c7f1-4bf4-9142-b5dfa63d58a5.dat
INFO:     127.0.0.1:34202 - "GET /api/histories/529fd61ab1c6cc36/contents/529fd61ab1c6cc36 HTTP/1.1" 404 Not Found
```

I think this shouldn't be a 404 and I have looked through the managers for the last 20 minutes trying to figure out what would cause a 404 and I don't see anything.  But also - it would be easy enough to swap the test to just assume the 404 means the dataset is purged.

--- a/lib/galaxy_test/base/populators.py
+++ b/lib/galaxy_test/base/populators.py
@@ -875,6 +875,9 @@ class BaseDatasetPopulator(BasePopulator):
 
         def _wait_for_purge():
             dataset_response = self._get(dataset_url)
+            # Accept 404 as an intermediate state - dataset will return 200 with purged=True later
+            if dataset_response.status_code == 404:
+                return None
             dataset_response.raise_for_status()
             dataset = dataset_response.json()
             return dataset.get("purged") or None

